### DISCUSSION TITLE
HTML rework / network configuration

### DISCUSSION THREAD

**Discussion by torei** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#164] [Ideas]
what do you think about...

.. new home screen:
![1_home](https://user-images.githubusercontent.com/6784876/147764432-218887d0-50f5-468c-b707-eb72bf43ba98.png)
.. new menu:
![2_home-menu](https://user-images.githubusercontent.com/6784876/147764434-d1eb82fd-42d0-4515-9d0d-3e45396049e9.png)
.. new spa config:
![3_spa-config](https://user-images.githubusercontent.com/6784876/147764438-f47d0923-7a9f-4839-99e4-3b9872d1f407.png)
... brand new network config:
![4_network-config](https://user-images.githubusercontent.com/6784876/147764441-eef6055f-5cda-4479-bb00-75df004d44e8.png)
.. new MQTT config:
![5_mqtt-config](https://user-images.githubusercontent.com/6784876/147764443-fc335960-e582-4349-a8ed-9b89244e0746.png)
.. nothing changed here :-)
![6_dir](https://user-images.githubusercontent.com/6784876/147764445-827cefc0-c72b-411a-a7a6-e70e90711e00.png)
.. new file uploader:
![7_file-uploader](https://user-images.githubusercontent.com/6784876/147764447-05fd1f63-c7b4-4cc1-8a74-e7be2b14ef36.png)
.. new file remover:
![8_file-remover](https://user-images.githubusercontent.com/6784876/147764450-52779dfc-cdeb-4b02-ac48-1b5c2428a91e.png)
.. new success screen for file operations:
![9_file-success](https://user-images.githubusercontent.com/6784876/147764452-d34798ee-7841-4358-bb6a-060b5f8cfc62.png)

please let me know what you guys - especially @visualapproach - think about it.
i did not push my local branch with all the changes.. yet.. 
sorry for editing tons of code - guess that will be a merge horror...

here is the "config.h" (former "credentials.h") just to let you take a look into my changes:

```cpp
#include <Arduino.h>
#include <ESP8266WiFi.h>

#define LEGACY_NAME "layzspa-test"

/*
 * Web Authentication
 */
/** get or set the state of the web authentication */
bool enableWebAuth = false;
/** get or set the username for web authentication */
const char *authUsername = "username";
/** get or set the password for web authentication */
const char *authPassword = "password";

/*
 * OTA Service Credentials
 */
/** get the name for the OTA service */
const char *OTAName = LEGACY_NAME;
/** get the password for the OTA service  */
const char *OTAPassword = "esp8266";

/*
 * Access Point Configuration
 */
/** get or set the state of the specific access point configuration */
bool enableAp = false;
/** get or set the name of the SSID */
String apSsid = "ssid";
/** get or set the password for the SSID */
String apPwd = "pwd";

/*
 * Static IP Configuration
 */
/** get or set the state of the static IP setup */
bool enableStaticIp4 = false;
/** get or set the IP address */
IPAddress ip4Address(192,168,0,30);
/** get or set the gateway address */
IPAddress ip4Gateway(192,168,0,1);
/** get or set the subnet mask */
IPAddress ip4Subnet(255,255,255,0);
/** get or set the primary DNS IP */
IPAddress ip4DnsPrimary(8,8,8,8);
/** get or set the secondary DNS IP */
IPAddress ip4DnsSecondary(8,8,4,4);

/*
 * MQTT Server Configuration
 */
/** get or set the state of the MQTT server connection */
bool enableMqtt = false;
/** get or set the MQTT server IP address */
IPAddress mqttIpAddress(192,168,0,20);
/** get or set the MQTT server port */
int mqttPort = 1883;
/** get or set the MQTT server username */
String mqttUsername = "username";
/** get or set the MQTT server password */
String mqttPassword = "password";
/** get or set the unique MQTT client ID */
String mqttClientId = LEGACY_NAME;
/** get or set the MQTT topic name */
String mqttBaseTopic = LEGACY_NAME;
```

**Comment by torei** [comment]
updated my dev branch:
https://github.com/torei/WiFi-remote-for-Bestway-Lay-Z-SPA/tree/develop

**Comment by visualapproach** [comment]
Hi @torei 
Thanks for contributing!
Overall it seems good. I like the menu feature most! It's useful for many people. I want to keep the logo on the main page though, in this repository.
Yeah, merging is scary. I guess you've added to the load/save-functions in the library, and to main.cpp. Anything else, except for the html-files? We can always make a test-pull to a new branch.

**Reply by torei** [reply]
alright, sounds good.
what about this:
![grafik](https://user-images.githubusercontent.com/6784876/147773196-8a7e84f1-209e-498c-83e3-4968a175b928.png)

i have kind of.. restructured the main.cpp. except for the data/* changes, the main.* and config.h only some smaller things

branch is updated, torei/develop

**Comment by visualapproach** [comment]
Great, I'll make a branch called "torei" which you can make a pull request to. And I'm working on a few updates myself now, so we'll see how it goes together. I'll try in a while.
PS I just browsed through your files a little and noticed a brightness multiplier. Didn't finish reading, but be aware it can only have a value 0 - 7. DS

**Comment by visualapproach** [comment]
Hi. I merged it into a temp branch. I replaced the "B" icon because I don't want "them" to claim infringement on logo. It wasn't the same, but still. Also I had to disable some wifi features. Probably changed some more, but it seems to work now. Thank you for the new menu, I think it's great.

**Reply by torei** [reply]
hey @visualapproach 
yeah i am following all changes currently because i have some updates.. merged your dev branch two times into mine.
just btw. sorry to tell, but you kind of messed up some git flow :D
why not merging the dev branch into a feature branch instead of creating new branches? would be much easier to manage.
i'm just testing my own dev branch. if you want, we could discuss changes in a PR? e.g. the password display thing

**Reply by torei** [reply]
oh and also.. you cannot blame any more because you merged all commits into one. so nobody is getting props on their code ;]

**Reply by torei** [reply]
just pushed all my changes into my "develop" branch and created a PR -> #169

**Reply by visualapproach** [reply]
> why not merging the dev branch into a feature branch instead of creating new branches?

1. Because I suck at git...

>  so nobody is getting props on their code

2. Doh! Unintentionally. see 1. That's why I didn't want to mess with the "original" branches, but played around in the "temp".
3. Maybe best to scrap the temp branch and do it from scratch again to preserve your contributions as yours.

**Reply by torei** [reply]
ok lol :D
nooo.. don't. just keep it. i'm good with it. and ofc this project is your baby and i'm happy to contribute.
i think you can delete the torei branch. all changes are still in my forked repo. i'm only working in the "develop" branch there

**Reply by visualapproach** [reply]
Ok, I'm done for today but we can look at it tomorrow or when convenient. Maybe you can give some advice how to handle your PR best. I'd like to merge everything down to master and development eventually, after testing. Thanks

**Reply by torei** [reply]
there are no conflicts in the PR. i have started to review my own changes and if you like, you can do some review on things as well.
otherwise you should be able to just merge or close the PR.
